src/lib/descriptor/mod.rs

```diff
@@ -14,69 +14,175 @@ mod cargo_alias;
 pub(crate) mod descriptor_deserializer;
 mod makefiles;
 
-use crate::io;
-use crate::plugin::descriptor::merge_plugins_config;
-use crate::scriptengine;
-use crate::types::{
-    Config, ConfigSection, EnvFile, EnvFileInfo, EnvValue, Extend, ExternalConfig, ModifyConfig,
-    Task,
-};
-use crate::version;
-use envmnt;
+use std::collections::HashSet;
+use std::env;
+use std::path::{Path, PathBuf};
+
 use fsio::path::as_path::AsPath;
 use fsio::path::from_path::FromPath;
 use indexmap::IndexMap;
-use std::env;
-use std::path::{Path, PathBuf};
-use toml;
+use once_cell::sync::Lazy;
+use petgraph::algo::{kosaraju_scc, toposort};
+use petgraph::graphmap::DiGraphMap;
+use petgraph::prelude::GraphMap;
+use regex::Regex;
+use {envmnt, toml};
+
+use crate::plugin::descriptor::merge_plugins_config;
+use crate::types::{
+    Config, ConfigSection, EnvFile, EnvFileInfo, EnvValue, EnvValueConditioned, EnvValueDecode,
+    EnvValuePathGlob, Extend, ExternalConfig, ModifyConfig, Task,
+};
+use crate::{io, scriptengine, version};
+
+static RE_VARIABLE: Lazy<Regex> = Lazy::new(|| Regex::new(r"\$\{.*}").unwrap());
+
+fn merge_env_depends_on_extract(val: &str) -> Vec<&str> {
```

Comment:
can we move the entire env loading/ordering logic to a new file? this file is big as it is